Exiled.Events/Patches/Events/Scp079/Pinging.cs

```diff
@@ -0,0 +1,74 @@
+// -----------------------------------------------------------------------
+// <copyright file="Pinging.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Scp079
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection.Emit;
+
+    using Exiled.API.Features;
+    using Exiled.Events.EventArgs.Scp079;
+    using HarmonyLib;
+    using Mirror;
+    using NorthwoodLib.Pools;
+    using PlayerRoles.PlayableScps.Scp079.Pinging;
+    using PlayerRoles.PlayableScps.Subroutines;
+    using RelativePositioning;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    ///     Patches <see cref="Scp079PingAbility.ServerProcessCmd" />.
+    ///     Adds the <see cref="PingingEventArgs" /> event for  SCP-079.
+    /// </summary>
+    [HarmonyPatch(typeof(Scp079PingAbility), nameof(Scp079PingAbility.ServerProcessCmd))]
+    internal static class Pinging
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+            int offset = 0;
+            int index = newInstructions.FindIndex(
+                instruction => instruction.Calls(Method(typeof(RelativePositionSerialization), nameof(RelativePositionSerialization.ReadRelativePosition)))) + offset;
+
+            // this._syncPos = ev.Position;
+            newInstructions.InsertRange(
+                index,
+                new CodeInstruction[]
+                {
+                    // new(OpCodes.Ldarg_0, 0),
+                    // new(OpCodes.Ldarg_1),
+                    new(OpCodes.Call, Method(typeof(Pinging), nameof(Pinging.ProcessPinging))),
+                    new(OpCodes.Ret),
+
+                    //To satisfy harmony but I should never get here.
```

Comment:
What do you mean?

Reply:
idk they are not from me

Reply:
Done - harmony apparently does not like

```
IL_005C: ldarg.0
IL_005D: ldarg.1
IL_005E: call       static System.Void Exiled.Events.Patches.Events.Scp079.Pinging::ProcessPinging(PlayerRoles.PlayableScps.Scp079.Pinging.Scp079PingAbility instance, Mirror.NetworkReader reader)
IL_0063: br =>      Label6
IL_0068: call       static RelativePositioning.RelativePosition RelativePositioning.RelativePositionSerialization::ReadRelativePosition(Mirror.NetworkReader reader)
```

Despite the fact that, br will always occur - I guess because it wants to complete the stack, which I get but it becomes unreachable code, I might as well wipe it from existence at that point but it's fine